VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -217,17 +217,18 @@ private  View getRowViewForLeaf(int position, View convertView, ViewGroup parent
         viewHolder.rowSubtitlePanel.setVisibility(View.GONE);
         viewHolder.bulkDownload.setVisibility(View.INVISIBLE);
 
-        if ( !unit.isMultiDevice() && unit.getType() != BlockType.VIDEO){
+        if (row.component instanceof VideoBlockModel) {
+            updateUIForVideo(position, convertView, viewHolder, row);
+        } else if (!unit.isMultiDevice() || unit.getType() == BlockType.VIDEO) {
+            // If we reach here & the type is VIDEO, it means the video is webOnly
```

Comment:
This logic seems wrong.. we should should the laptop icon _whenever_ the `multiDevice` field is `false`, whether the type is video or not. So just remove the `&& unit.getType() != BlockType.VIDEO` from the previous logic.